lib/io/IO.js

```diff
@@ -0,0 +1,72 @@
+import { NoteModel } from "../note";
+import { insertNote } from "./sqlite/insert.js";
+import { queryNotes } from "./sqlite/query.js";
+import { cascadeDeleteNote } from "./sqlite/delete.js";
+import { auroraDirContext, saveToAsync, readFromAsync } from "./util.js";
+
+const preferencesFile = "aurora-preferences.json";
+/*
+  Saves the specified note. Overwrites an existing note with matching id.
+  note : NoteModel
+  onSuccess() : function invoked if successfully saved.
+  onFailure(err) : function invoked if failure to save. Takes in error object.
+   */
+function saveNote(note, onSuccess, onFailure) {
+  if (!(note instanceof NoteModel)) {
```

Comment:
Consider moving this into its own function to avoid repetitive stuff? Normally it's not a huge issue since it's just twice, but since there's a specific error message, I would do it. 

``` js
function throwIfNotNoteModel(Note) {
  if (!(note instanceof NoteModel)) {
    throw new Error("Attempted to save something that is not a NoteModel.");
  }
}
```

Reply:
good point.